RFS/src/main/java/com/rfs/version_es_6_8/ShardMetadataData_ES_6_8.java

```diff
@@ -108,8 +107,7 @@ public long getTotalSizeBytes() {
 
     @Override
     public List<ShardFileInfo> getFiles() {
-        List<ShardFileInfo> convertedFiles = new ArrayList<>(files);
-        return convertedFiles;
+        return new ArrayList<>(files);
```

Comment:
If files is meant to be immutable, you could just make a @Getter on the field and wrap line 60 in a toUnmodifiableList or something similar.

Reply:
TBH, all of these classes are a bit of an inconsistent mess in terms of interface and implementation.  I'll try to take a quick pass on them and see if I can improve things a bit.